internal/armada/server/lease.go

```diff
@@ -386,6 +371,12 @@ func (q *AggregatedQueueServer) getJobs(ctx context.Context, req *api.StreamingL
 		if skipNode {
 			continue
 		}
+
+		// Record the which node each job is scheduled on. Necessary for gang preemption.
+		for _, job := range jobs {
+			nodeIdByJobId[job.Id] = node.Id
```

Comment:
Not populating this was the reason gang preemption didn't work correctly previously.